.github/workflows/productized.yaml

```diff
@@ -0,0 +1,158 @@
+name: Productized CI
+
+on:
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  main:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest, macos-13, windows-latest ]
+        version: [ max ] # [ "x.x.x" | latest | max ]
+        type: [ stable ] # [ stable | insider ]
+        camelversion: [ "4.0.0",  "4.0.0.redhat-00031"]
+
+    env:
+      CODE_VERSION: ${{ matrix.version }}
+      CODE_TYPE: ${{ matrix.type }}
+      TEST_RESOURCES: test-resources
+      CAMEL_VERSION: ${{ matrix.camelversion }}
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: 17
+          distribution: "temurin"
+
+      - name: Test java
+        run: java --version
+
+      - name: Install JBang (ubuntu, macOS)
+        if: matrix.os == 'ubuntu-latest' || matrix.os == 'macos-13'
+        run: |
+          curl -Ls https://sh.jbang.dev | bash -s - app setup
+          echo "$HOME/.jbang/bin" >> $GITHUB_PATH
+
+      - name: Install JBang (windows)
+        if: matrix.os == 'windows-latest'
+        run: choco install jbang
+
+      - name: Setup JBang (trusted sources)
+        run: jbang trust add https://github.com/apache/
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18.15.x
+          cache: npm
+
+      - name: Setup vsce
+        run: |
+          npm install -g typescript @vscode/vsce
+          vsce --version
+
+      - name: npm-ci
+        run: npm ci
+
+      - name: npm-vscode:prepublish
+        run: npm run vscode:prepublish
+
+      - name: test (ubuntu)
+        id: test_Ubuntu
+        if: matrix.os == 'ubuntu-latest'
+        run: xvfb-run --auto-servernum npm test
+
+      - name: test (macos, windows)
+        if: matrix.os != 'ubuntu-latest'
+        run: npm test
+
+      - name: ui test (ubuntu)
+        id: uiTest_Ubuntu
+        if: matrix.os == 'ubuntu-latest'
+        run: xvfb-run --auto-servernum npm run ui-test
+
+      - name: ui test (macos, windows)
+        id: uiTest_MacOS_Windows
+        if: matrix.os != 'ubuntu-latest'
+        run: npm run ui-test
+
+      - name: vsce-package
+        run: vsce package
+
+      - name: get-npm-version
+        id: package-version
+        if: matrix.os == 'ubuntu-latest'
+        uses: martinbeentjes/npm-get-version-action@v1.3.1
```

Comment:
this can be removed, it is related to sbom which is not generated in this pipeline